walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,189 @@
+        Collections.sort(busySchedule, TimeRange.ORDER_BY_END);
+
+        //time nodes that are contained in others
+        trimContainedNodes(busySchedule);
+
+        
+
+        //test
+        for(int k=0 ; k < busySchedule.size(); k++){
+
+            TimeRange currentTimeRange = busySchedule.get(k);
+            int start = currentTimeRange.start();
+            int end = currentTimeRange.end();
+            int duration = currentTimeRange.duration();
+
+            if(k == 0 && busySchedule.size()!=0){
+        
+                TimeRange possibleTime = TimeRange.fromStartEnd(dayStart,start,false);
+                
+                        if(possibleTime.duration() >= meetingDuration){
+                        possibleTimes.add(possibleTime);
+                        
+                        
+                    }
+                
+            }
+
+            if(k == busySchedule.size()-1 && busySchedule.size()!=0){
+                
+                TimeRange possibleTime = TimeRange.fromStartEnd(end,dayEnd,true);
+
+                if(k!=0){
+                    int possibleStartTime = Math.max(end,possibleTime.end());
+                    TimeRange anotherPossibleTime = TimeRange.fromStartEnd(end,dayEnd,true);
+
+                    if(anotherPossibleTime.overlaps(possibleTime)){
+                        if(anotherPossibleTime.duration() >= meetingDuration){
+                            possibleTimes.add(anotherPossibleTime);
+                        
+                        }
+
+                    }
+                    
+                }else{
+                    if(possibleTime.duration() >= meetingDuration){
+                        possibleTimes.add(possibleTime); 
+                        
+                    }
+                }     
+            }
+
+            if(k+1<busySchedule.size()){
+                
+                TimeRange possibleTime = TimeRange.fromStartEnd(end,busySchedule.get(k+1).start(),false);
+                
+                    if(possibleTime.duration() >= meetingDuration){
+                    possibleTimes.add(possibleTime);
+
+                    
+                    
+                }
+            }
+
+        }
+        
+
+
+    
+     return possibleTimes;
   }
+
+    //check if person is attendee of event
+   public boolean isAnAttendee(String attendee, Event event){
+       if (event.getAttendees().contains(attendee)){
+            return true;
+       }        
+       else{
+            return false;
+       }
+                
+   }
+    //adds event to an attendee schedule. 
+   public void addToSchedule(HashSet<TimeRange> attendeeSchedule, Event event){
+       attendeeSchedule.add(event.getWhen());
+   }
+    //check if there is an overlap between a meeting time and an attendee schedule
+   public boolean checkIfOverlap(TimeRange currentMeetingTime,HashSet<TimeRange> attendeeSchedule){
+
+     Iterator<TimeRange> i = attendeeSchedule.iterator(); 
```

Comment:
Similar comment to above, I would only use an `Iterator` if you need to potentially remove values while you iterate. If not, I would use an enhanced for.